helpers/embeds.js

```diff
@@ -0,0 +1,102 @@
+import { html, nothing, render } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { map } from 'lit/directives/map.js';
+import { tryGetPluginByKey } from './plugins.js';
+
+const embedTypeAttributeName = 'data-d2l-embed-type';
+
+function tryGetEmbedRendererPlugin(embedType) {
+	const embedRendererPlugin = tryGetPluginByKey('d2l-html-embed-renderer', embedType);
+	if (!embedRendererPlugin) {
+		console.warn(`d2l-html-embed-renderer: Can't find plugin for ${embedType} embed type.`);
+	}
+
+	return embedRendererPlugin;
+}
+
+export async function createEmbedViewPlaceholder(embedType, props) {
+	const embedRendererPlugin = tryGetEmbedRendererPlugin(embedType);
+	if (!embedRendererPlugin) return;
+
+	const placeholderData = await embedRendererPlugin.getViewPlaceholderData(props);
+	const contents = placeholderData.contents
+		? map(Object.entries(placeholderData.contents), ([id, content]) => html`<template data-d2l-embed-template-id=${id}>${content}</template>`)
```

Comment:
I think this is safe to do in the view placeholder because (1) we're templating the content anyways, and (2) if there's user-authored HTML in here, we're expecting it to be pre-filtered. This doesn't have a means to support more complex structures in terms of the placeholders, but honestly, I'm not sure it's worth spending too much time exploring that before we need it. We can always add something like that later.